src/main/java/au/org/ala/biocache/dao/SearchDAOImpl.java

```diff
@@ -717,11 +852,12 @@ public void writeCoordinatesToStream(SearchRequestParams searchParams,OutputStre
      * @param includeSensitive
      * @throws Exception
      */
-    public Map<String, Integer> writeResultsFromIndexToStream(final DownloadRequestParams downloadParams,
+    public ConcurrentMap<String, AtomicInteger> writeResultsFromIndexToStream(final DownloadRequestParams downloadParams,
                                                                          OutputStream out,
                                                                          boolean includeSensitive, final DownloadDetailsDTO dd, boolean checkLimit) throws Exception {
+        ExecutorService nextExecutor = getSolrThreadPoolExecutor();
```

Comment:
getSolrThreadPoolExecutor() isn't used by DownloadControlThread (which I was under the impression takes all of the offline downloads under the current pattern). Each of the thread pools that come in through the concurrent.downloads.json configuration get their own ExecutorService that is preconfigured and encapsulated in DownloadControlThread. The current design relies on the ExecutorService behind each DownloadControlThread being setup as a fixed thread pool so would need to rethink that design using another pattern if necessary.